WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/TodayWidgetUpdater.kt

```diff
@@ -0,0 +1,123 @@
+package org.wordpress.android.ui.stats.refresh.lists.widget
+
+import android.appwidget.AppWidgetManager
+import android.content.ComponentName
+import android.content.Context
+import android.widget.RemoteViews
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.fluxc.store.stats.insights.TodayInsightsStore
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import org.wordpress.android.ui.stats.StatsTimeframe
+import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureFragment.ViewType
+import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureViewModel.Color
+import org.wordpress.android.ui.stats.refresh.utils.toFormattedString
+import org.wordpress.android.util.NetworkUtilsWrapper
+import org.wordpress.android.viewmodel.ResourceProvider
+import javax.inject.Inject
+
+class TodayWidgetUpdater
+@Inject constructor(
+    private val appPrefsWrapper: AppPrefsWrapper,
+    private val siteStore: SiteStore,
+    private val networkUtilsWrapper: NetworkUtilsWrapper,
+    private val resourceProvider: ResourceProvider,
+    private val todayInsightsStore: TodayInsightsStore,
+    private val widgetUtils: WidgetUtils
+) : WidgetUpdater {
+    override fun updateAppWidget(
+        context: Context,
+        appWidgetManager: AppWidgetManager,
+        appWidgetId: Int
+    ) {
+        val showColumns = widgetUtils.isWidgetWiderThanLimit(appWidgetManager, appWidgetId)
+        val colorMode = appPrefsWrapper.getAppWidgetColor(appWidgetId) ?: Color.LIGHT
+        val siteId = appPrefsWrapper.getAppWidgetSiteId(appWidgetId)
+        val siteModel = siteStore.getSiteBySiteId(siteId)
+        val networkAvailable = networkUtilsWrapper.isNetworkAvailable()
+        val views = RemoteViews(context.packageName, widgetUtils.getLayout(showColumns, colorMode))
+        views.setTextViewText(R.id.widget_title, resourceProvider.getString(R.string.stats_insights_today_stats))
+        widgetUtils.setSiteIcon(siteModel, context, views, appWidgetId)
+        siteModel?.let {
+            views.setOnClickPendingIntent(
+                    R.id.widget_title, widgetUtils.getPendingSelfIntent(
+                    context,
+                    siteModel.id,
+                    StatsTimeframe.INSIGHTS
+            )
+            )
+        }
+        if (networkAvailable && siteModel != null) {
+            if (showColumns) {
+                views.setOnClickPendingIntent(
+                        R.id.widget_content,
+                        widgetUtils.getPendingSelfIntent(context, siteModel.id, StatsTimeframe.INSIGHTS)
+                )
+                showColumns(appWidgetManager, appWidgetId, views, siteModel)
+            } else {
+                widgetUtils.showList(
```

Comment:
Do we need to `setOnClickPendingIntent` for a list view? It looks like that currently list view just opens Stats without sending any data with intent.

Reply:
I'll check with Sylvester about what we need to do on item click in the list. It's likely we can only use the `setOnClickPendingIntent` to open the stats and we don't need individual item click 👍